lib/genesis/helpers.go

```diff
@@ -232,33 +336,249 @@ func buildRawArrayInterface(a []interface{}, kv *keyValue) error {
+	if k := reflect.Indirect(val).Type().Field(0).Name; k == "CurrentSchedule" {
+		key, err = generateStorageKey(prefixKey, k)
+		if err != nil {
+			return err
+		}
+	}
+
+	encode, err := scale.Marshal(c)
+	if err != nil {
+		return err
+	}
+	res[key] = common.BytesToHex(encode)
+	return nil
+}
+
+func generateKeyValue(s interface{}, prefixKey string, res map[string]string) error {
+	val := reflect.ValueOf(s)
+	n := reflect.Indirect(val).NumField()
+	for i := 0; i < n; i++ {
+		val := reflect.ValueOf(s)
+		storageKey := reflect.Indirect(val).Type().Field(i).Name
+		if storageKey == "Phantom" { //TODO: figure out what to do with Phantom as its value is null
+			continue
+		}
+		key, err := generateStorageKey(prefixKey, storageKey)
+		if err != nil {
+			return err
+		}
+		value, err := generateStorageValue(s, i)
+		if err != nil {
+			return err
+		}
+		res[key] = common.BytesToHex(value)
+	}
+	return nil
+}
+
 func formatKey(kv *keyValue) (string, error) {
 	switch {
-	case reflect.DeepEqual([]string{"grandpa", "authorities"}, kv.key):
+	case reflect.DeepEqual([]string{"Grandpa", "authorities"}, kv.key):
 		kb := []byte(`:grandpa_authorities`)
 		return common.BytesToHex(kb), nil
-	case reflect.DeepEqual([]string{"system", "code"}, kv.key):
+	case reflect.DeepEqual([]string{"System", "changesTrieConfig", "code"}, kv.key):
 		kb := []byte(`:code`)
 		return common.BytesToHex(kb), nil
 	default:
 		if len(kv.key) < 2 {
 			return "", errors.New("key array less than 2")
 		}
-		prefix, err := common.Twox128Hash([]byte(kv.key[0]))
-		if err != nil {
-			return "", err
-		}
-		keydata, err := common.Twox128Hash([]byte(kv.key[1]))
-		if err != nil {
-			return "", err
+
+		return generateStorageKey(kv.key[0], kv.key[1])
+	}
+}
+
+func generateSessionKeyValue(s *session, prefixKey string, res map[string]string) error {
+	val := reflect.ValueOf(s).Elem()
+	moduleName, err := common.Twox128Hash([]byte(prefixKey))
+	if err != nil {
+		return err
+	}
+
+	storageVal, ok := val.Field(0).Interface().([][]interface{})
+	if !ok {
+		return nil
+	}
+
+	for _, strV := range storageVal {
+		for _, v := range strV {
+			switch t := v.(type) {
+			case string:
+				var nextKeyHash []byte
+				nextKeyHash, err = common.Twox128Hash([]byte("NextKeys"))
+				if err != nil {
+					return err
+				}
+
+				acc := crypto.PublicAddressToByteArray(common.Address(t))
+				var accIDHash []byte
+				accIDHash, err = common.Twox64(acc)
+				if err != nil {
+					return err
+				}
+
+				res[common.BytesToHex(append(append(moduleName, nextKeyHash...), append(accIDHash, acc...)...))] = common.BytesToHex(acc)
+			case map[string]interface{}:
+				var storagePrefixKey []byte
+				storagePrefixKey, err = common.Twox128Hash([]byte("KeyOwner"))
+				if err != nil {
+					return err
+				}
+
+				storagePrefixKey = append(moduleName, storagePrefixKey...)
+				validatorAccID := crypto.PublicAddressToByteArray(common.Address("5GNJqTPyNqANBkUVMN1LPPrxXnFouWXoe2wNSmmEoLctxiZY"))
```

Comment:
who's address is this / where is it from?

Reply:
Removed. I hard-coded it for testing.